### PR TITLE
Updates for Swift 3

### PR DESCRIPTION
**Pull request by bensyverson**
Updated syntax for Swift 3 (build swift-DEVELOPMENT-SNAPSHOT-2016-03-24-a)

### CODE REVIEW

**Comment by Danappelxx** [comment]
Awesome! There's some extra whitespace in some areas (I'll leave comments there) but other than that, LGTM and thanks!

File: Sources/BinaryJSON/Error.swift [hunk 1]
```diff
@@ -35,21 +35,20 @@ public extension BSON {
             
             assert(unsafePointer != nil, "Trying to create Error from nil pointer")
             
-            var messageTuple = unsafePointer.memory.message
+            var messageTuple = unsafePointer.pointee.message
             
             let message = withUnsafePointer(&messageTuple) { (unsafeTuplePointer) -> String in
                 
-                let charPointer = unsafeBitCast(unsafeTuplePointer, UnsafePointer<CChar>.self)
+				let charPointer = unsafeBitCast(_: unsafeTuplePointer, to: UnsafePointer<CChar>.self)
```
Comment (Danappelxx) [inline Sources/BinaryJSON/Error.swift]: whitespace

File: Sources/BinaryJSON/UnsafePointer.swift [hunk 2]
```diff
@@ -271,15 +269,13 @@ private extension BSON {
...
-                let bufferPointer = UnsafeMutablePointer<UnsafePointer<UInt8>>.alloc(1)
-                
-                defer { bufferPointer.dealloc(1) }
-                
+                let bufferPointer = UnsafeMutablePointer<UnsafePointer<UInt8>>(allocatingCapacity:1)
+				
                 bson_iter_binary(&iterator, &subtype, &length, bufferPointer)
                 
-                var bytes: [UInt8] = [UInt8](count: Int(length), repeatedValue: 0)
+				var bytes: [UInt8] = [UInt8](repeating: 0, count: Int(length))
```
Comment (Danappelxx) [inline Sources/BinaryJSON/UnsafePointer.swift]: whitespace

File: Sources/BinaryJSON/UnsafePointer.swift [hunk 3]
```diff
@@ -370,17 +364,17 @@ private extension BSON {
                 
                 var scopeLength: UInt32 = 0
                 
-                let scopeBuffer = UnsafeMutablePointer<UnsafePointer<UInt8>>.alloc(1)
+                let scopeBuffer = UnsafeMutablePointer<UnsafePointer<UInt8>>(allocatingCapacity:1)
                 
-                defer { scopeBuffer.destroy(); scopeBuffer.dealloc(1) }
+                defer { scopeBuffer.deinitialize(); /*scopeBuffer.deallocCapacity(1) */ }
```
Comment (Danappelxx) [inline Sources/BinaryJSON/UnsafePointer.swift]: Is there a reason why this commented out?

**Comment by bensyverson** [comment]
BTW, since Swift 3 is still so young, feel free to hold off on the merge and/or create a Swift 3 branch

File: Sources/BinaryJSON/UnsafePointer.swift [hunk 4]
```diff
@@ -366,7 +366,7 @@ private extension BSON {
                 
                 let scopeBuffer = UnsafeMutablePointer<UnsafePointer<UInt8>>(allocatingCapacity:1)
                 
-                defer { scopeBuffer.deinitialize(); /*scopeBuffer.deallocCapacity(1) */ }
+				defer { scopeBuffer.deinitialize(); /*scopeBuffer.deallocCapacity(1) */ }
```
Comment (Danappelxx) [inline Sources/BinaryJSON/UnsafePointer.swift]: I think my comment here still applies, plus you messed up the whitespace again :stuck_out_tongue:

**Comment by Danappelxx** [comment]
Nah, I'm all for being on the bleeding edge - I've been meaning to do this so that it would work with Zewo and others but didn't get around to it.

File: Sources/BinaryJSON/UnsafePointer.swift (hunk 4 above)
Comment (bensyverson) [inline Sources/BinaryJSON/UnsafePointer.swift]: Gah! Must be a space vs tab thing

File: Sources/BinaryJSON/UnsafePointer.swift (hunk 3 above)
Comment (bensyverson) [inline Sources/BinaryJSON/UnsafePointer.swift]: dealloc and deallocCapacity apparently don't apply to that data type anymore. I'm going to look into it, but I think deinitialize() might do the trick here.

Comment (Danappelxx) [inline Sources/BinaryJSON/UnsafePointer.swift]: Ah, it was renamed to `deallocateCapacity`. Fix that and I'll merge :smiley:

Comment (bensyverson) [inline Sources/BinaryJSON/UnsafePointer.swift]: Ah, that did work! Thanks!

File: Sources/BinaryJSON/Context.swift [hunk 5]
```diff
@@ -93,21 +93,21 @@ public protocol BitMaskOption: RawRepresentable {
...
     
-    func optionsBitmask() -> Self.Generator.Element.RawValue {
+    func optionsBitmask() -> Self.Iterator.Element.RawValue {
         
         let array = self.filter { (_) -> Bool in return true }
         
-        return Self.Generator.Element.optionsBitmask(array)
+        return Self.Iterator.Element.optionsBitmask(array)
     }
 }
```
Comment (Danappelxx) [inline Sources/BinaryJSON/Context.swift]: Could you do me favor and insert a new line at the end of the file? I just need you to push a commit to trigger travis one last time.

Comment (bensyverson) [inline Sources/BinaryJSON/Context.swift]: Sure! :+1:

**Comment by Danappelxx** [comment]
Looks like `swift test` doesn't work for some reason... I looked into it but have come up blank. Anyway, I can't think of a reason that this would break anything so I'm merging it.

Thanks!

**Comment by bensyverson** [comment]
Cool! Yeah, I couldn't get the tests to run either—but it seems to be working in conjunction with SwiftMongoDB (I was able to connect and insert documents).

**Pull request merged**